source/features/notifications-filter.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+import {CheckIcon, CheckCircleIcon, XCircleIcon, IssueOpenedIcon, GitPullRequestIcon, GitPullRequestDraftIcon, GitMergeIcon, DotIcon, DotFillIcon} from '@primer/octicons-react';
+
+import features from '.';
+
+const filters = {
+	PR: '[href*="/pull/"]',
+	Issue: '[href*="/issues/"]',
+	Open: ':is(.octicon-issue-opened, .octicon-git-pull-request)',
+	Closed: ':is(.octicon-issue-closed, .octicon-git-pull-request-closed)',
+	Draft: '.octicon-git-pull-request-draft',
+	Merged: '.octicon-git-merge',
+	Read: 'notification-read',
+	Unread: 'notification-unread'
+};
+
+type Filter = keyof typeof filters;
+type Category = 'Type' | 'Status' | 'Read';
+
+function handleSelection({target}: Event): void {
+	const activeFilters: Record<Category, string[]> = {
```

Comment:
Do you need the category here? `is:read` should be identical to `is:pr` internally, right? They're all just matched against the selectors in `filters`.

Could this whole function be shortened to something like:

```js
// Merge all the filters into a single selector
const filteredNotificationSelector = select
	.all('[aria-checked="true"]', target as Element)
	.map(el => el.value)
	.join(''); 


for (const notification of select.all('.notifications-list-item')) {
	const hidden = notification.matches(filteredNotificationSelector);
	notification.hidden = hidden;
	select('.js-notification-bulk-action-check-item').toggleAttribute('data-check-all-item', hidden)
}

// Hide empty notifications groups
for (const group of select.all('.js-notifications-group')) {
	group.hidden = !select.exists('.notifications-list-item:not([hidden])', group);
}
```

Reply:
The filtering logic I've chosen to implement is as follow:
- filters in the same category are combined with OR
- categories are combined with AND

I don't think this can be expressed with a single selector. (Plus I've got to use `.matches()` for the Read/Unread filters since `querySelector()` doesn't look at the root.)